### PR TITLE
chore(test): Add README.me, CONTRIBUTOR update

### PR DESCRIPTION
**Pull request by TheLarkInn**
This PR adds a README.md inside of our test suite.

**What kind of change does this PR introduce?**

Repo Documentation
**Did you add tests for your changes?**

**If relevant, link to documentation update:**
N/A

**Summary**
People seem to not quite yet understand how to add tests and work through the test suite in the webpack code. I think a small rundown will benefit the quality of tests and PR's submitted.

**Does this PR introduce a breaking change?**

**Other information**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -48,7 +48,7 @@ your pull request should be accepted quickly.
 
 Some things that will increase the chance that your pull request is accepted:
 
-* Write tests
+* [Write tests](./test/README.m`d)
```
Comment (Equinox) [inline CONTRIBUTING.md]: I don't think the "`" was intended here.

Comment (TheLarkInn) [inline CONTRIBUTING.md]: Certainly not!! Thanks

**Comment by Equinox** [comment]
@TheLarkInn There are a few small errors (one missing "`", a spelling error): is it better to point them out now or PR a correction after merge?

**Comment by TheLarkInn** [comment]
Yes please feel free and mark them!!

File: test/README.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+To add a new case, create a new directory inside of the top level test groups, and then add an `index.js` file (and any other supporting files). 
+
+By default this file will be the entry point for the test suite and you can add your `it()`'s there. This will also become bundled so that node env support happens as well.  
+
+#### configCases (`ConfigTestCases.test.js`)*
+If you are trying to solve a bug which is reproducible when x and y properties are used together in a config, then configCases is the place to be!!!! 
+
+In addition to an `index.js`, these configCases require a `webpack.config.js` is located inside of your test suite. This will run this specific config through `webpack` just as you were building individually. They will use the same loading/bundling techniqe of your `it()` tests, however you now have a more specific config use cases that you can write even before you start coding.
```
Comment (Equinox) [inline test/README.md]: techniqe > technique

File: test/README.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+* If the actual output matches `expected.txt`, the tests passes and you are free to submit that PR with pride!!!
+
+## Questions? Comments? 
+If you are still nervous or don't quite understand, please submit an issue and tag us in it, and provide a relevant PR while working on!
+
+
+## Footnotes
+* webpack's parser supports the use of ES2015 features like arrow functions, harmony exports, etc. However as a library we follow NodeJS's timeline for dropping older versions of node. Because of this we expect your tests on Travis to pass all the way back to NodeJS v0.12; Therefore if you would like specific tests that use these features to be ignored if they are not supported, then you should add a `test.filter.js` file. This allows you to import the syntax needed for that test, meanwhile ignoring it on node versions (during CI) that don't support it. webpack has a variety of hel exapmles you can refer to if you are just starting out. See the `./helpers` foler to find a list of the verion
```
Comment (Equinox) [inline test/README.md]: hel > help
foler > folder
verion > version(s?).

File: test/README.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+By default this file will be the entry point for the test suite and you can add your `it()`'s there. This will also become bundled so that node env support happens as well.  
+
+#### configCases (`ConfigTestCases.test.js`)*
+If you are trying to solve a bug which is reproducible when x and y properties are used together in a config, then configCases is the place to be!!!! 
+
+In addition to an `index.js`, these configCases require a `webpack.config.js` is located inside of your test suite. This will run this specific config through `webpack` just as you were building individually. They will use the same loading/bundling techniqe of your `it()` tests, however you now have a more specific config use cases that you can write even before you start coding. 
+
+#### statsCases (`Stats.test.js`)
+Stats cases are similar to configCases except specifically focusing on the `expected` output of your stats. Instad of writing to the console, however the output of stats will be written to disk.
```
Comment (Equinox) [inline test/README.md]: Instad > Instead

File: test/README.md (hunk 3 above)
Comment (TheLarkInn) [inline test/README.md]: Thank you very much, I admittedly was very sleepy when I pushed this so I'm grateful for the extra spelling checks.

**Pull request merged**